src/DeviceHardDrive.php

```diff
@@ -215,6 +229,7 @@ public function getHTMLTableCellForItem(
                     );
                 }
 
+                DeviceHardDriveType::getHTMLTableCellsForItem($row, $this, null, $options);
```

Comment:
In `DeviceMemory`, it is done this way.

```suggestion
                if ($this->fields["deviceharddrivetypes_id"]) {
                    $row->addCell(
                        $row->getHeaderByName('deviceharddriver_type'),
                        Dropdown::getDropdownName(
                            "glpi_deviceharddrivetypes",
                            $this->fields["deviceharddrivetypes_id"]
                        ),
                        $father
                    );
                }
```